File: Forca/forca.js
function jogoDaForca() {
    const opcoes = ["emprego", "estagio", "desenvolvedor", "analista", "coisa", "nenhuma"];
    const palavraSecreta = opcoes[Math.floor(Math.random() * opcoes.length)].toLowerCase();
    const letrasCertas = new Set();
    const letrasErradas = new Set();
    const tentativasMaximas = 6;

    alert("Vamos jogar!");

    while (true) {
        let palavraMostrada = "";
        for (let letra of palavraSecreta) {
            if (letrasCertas.has(letra)) {
                palavraMostrada += letra + " ";
            } else {
                palavraMostrada += "_ ";
            }
        }
        console.log("\nPalavra:", palavraMostrada.trim());

        if (letrasErradas.size > 0) {
            console.log("Letras erradas:", Array.from(letrasErradas).join(" "));
        }

        if ([...palavraSecreta].every(letra => letrasCertas.has(letra))) {
            console.log("\nParabéns! Você acertou a palavra:", palavraSecreta);
            break;
        }

        if (letrasErradas.size >= tentativasMaximas) {
            console.log("\nGame Over! A palavra era:", palavraSecreta);
            break;
        }

        let tentativa = "";
        while (true) {
            tentativa = prompt("Digite uma letra:").toLowerCase();
            if (!tentativa || tentativa.length !== 1 || !tentativa.match(/[a-z]/i)) {
                alert("Por favor, digite apenas uma letra válida.");
            } else if (letrasCertas.has(tentativa) || letrasErradas.has(tentativa)) {
                alert("Você já tentou essa letra. Tente outra.");
            } else {
                break;
            }
        }

        if (palavraSecreta.includes(tentativa)) {
            letrasCertas.add(tentativa);
        } else {
            letrasErradas.add(tentativa);
            alert(`Errou! Você tem ${tentativasMaximas - letrasErradas.size} tentativas restantes.`);
        }
    }
}

jogoDaForca();

File: Forca/forca.py
import random

def jogo_da_forca():
    opcoes = ["emprego", "estagio", "desenvolvedor", "analista", "coisa", "nenhuma"]
    
    
    palavra_secreta = random.choice(opcoes).lower()
    letras_certas = set()
    letras_erradas = set()
    tentativas = 6 
    
    print("Vamos jogar!f")
    
    while True:
        palavra_mostrada = ""
        for letra in palavra_secreta:
            if letra in letras_certas:
                palavra_mostrada += letra + " "
            else:
                palavra_mostrada += "_ "
        print("\nPalavra:", palavra_mostrada)
        
        if letras_erradas:
            print("Letras erradas:", " ".join(letras_erradas))
        
        if all(letra in letras_certas for letra in palavra_secreta):
            print("\nParabéns! Você acertou a palavra:", palavra_secreta)
            break
        
        if len(letras_erradas) >= tentativas:
            print("\nGame Over! A palavra era:", palavra_secreta)
            break
        
        while True:
            tentativa = input("Digite uma letra: ").lower()
            if len(tentativa) != 1 or not tentativa.isalpha():
                print("Por favor, digite apenas uma letra válida.")
            elif tentativa in letras_certas or tentativa in letras_erradas:
                print("Você já tentou essa letra. Tente outra.")
            else:
                break
        
        if tentativa in palavra_secreta:
            letras_certas.add(tentativa)
        else:
            letras_erradas.add(tentativa)
            print(f"Errou! Você tem {tentativas - len(letras_erradas)} tentativas restantes.")

if __name__ == "__main__":
    jogo_da_forca()